dev/kyuubi-extension-spark-3-1/src/main/scala/org/apache/spark/sql/execution/ZorderOrdering.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, BaseOrdering, SortOrder}
+import org.apache.spark.sql.types.{BinaryType, BooleanType, ByteType, DoubleType, FloatType, IntegerType, LongType, ShortType, StringType, StructType}
+
+class ZorderOrdering(ordering: Seq[SortOrder],
+                      schema: StructType) extends BaseOrdering {
```

Comment:
Can we make a new `Expression` which return the binary type value and produced by Z-order ?

IMO we can apply Z-order with Spark `Ordering` simply, like `Sort(zorder(c1, c2, c3), ...)` instead. We can still name it as `zorder by c1, c2, c3` in `.g4`. The benifits of this approach is:

* more cleaner since less code copy
* support global z-order, otherwise we need to add a new partitioner

Reply:
OK, It is more concise to use zorder UDF compared with current solution.